contracts/adapters/PrtFeeSplitExtension.sol

```diff
@@ -0,0 +1,224 @@
+import { PreciseUnitMath } from "../lib/PreciseUnitMath.sol";
+
+/**
+ * @title PrtFeeSplitExtension
+ * @dev Extension that allows for splitting and setting streaming and mint/redeem fees with a 
+ * PRT Staking Pool. The operator can accrue fees from the streaming fee module and distribute
+ * them to the operator and the PRT Staking Pool, snapshotting the PRT Staking Pool. The operator 
+ * can update the PRT staking pool address and the fee split between the operator and the 
+ * PRT staking pool. Includes an optional allow list and timelock on accrue function.
+ */
+contract PrtFeeSplitExtension is FeeSplitExtension {
+    using Address for address;
+    using PreciseUnitMath for uint256;
+    using SafeMath for uint256;
+
+    /* ============ Events ============ */
+
+    event AnyoneAccrueUpdated(bool isAnyoneAllowedToBid);
+    event AccruerStatusUpdated(address indexed accruer, bool isBidderAllowed);
+    event OperatorFeeSplitUpdated(uint256 newFeeSplit);
+    event PrtFeesDistributed(
+        address indexed operatorFeeRecipient,
+        address indexed prtStakingPool,
+        uint256 operatorTake,
+        uint256 prtTake
+    );
+    event PrtStakingPoolUpdated(address newPrtStakingPool);
+
+    /* ============ Immutables ============ */
+
+    IPrt public immutable prt;
+
+    /* ============ State Variables ============ */
+
+    bool public isAnyoneAllowedToAccrue;
+    address[] accruersHistory;
+    mapping(address => bool) accrueAllowList;
+    IPrtStakingPool public prtStakingPool;
+
+    /* ============ Modifiers ============ */
+
+    modifier onlyAllowedAccruer() {
+        require(_isAllowedAccruer(msg.sender), "Not allowed to accrue");
+        _;
+    }
+
+    /* ============ Constructor ============ */
+
+    constructor(
+        IBaseManager _manager,
+        IStreamingFeeModule _streamingFeeModule,
+        IIssuanceModule _issuanceModule,
+        uint256 _operatorFeeSplit,
+        address _operatorFeeRecipient,
+        IPrt _prt
+    )
+        public
+        FeeSplitExtension(
+            _manager,
+            _streamingFeeModule,
+            _issuanceModule,
+            _operatorFeeSplit,
+            _operatorFeeRecipient
+        )
+    {
+        require(_prt.setToken() == address(manager.setToken()), "SetToken mismatch with Prt");
+        prt = _prt;
+    }
+
+    /* ============ External Functions ============ */
+
+    /**
+     * @notice MUTUAL UPGRADE: Updates PRT staking pool. PRT staking pool must have this extension set as the feeSplitExtension.
+     * @param _prtStakingPool Address of the new PRT staking pool
+     */
+    function updatePrtStakingPool(IPrtStakingPool _prtStakingPool) 
+        external 
+        mutualUpgrade(manager.operator(), manager.methodologist()) 
+    {
+        require(address(_prtStakingPool) != address(0), "Zero address not valid");
+        require(_prtStakingPool.feeSplitExtension() == address(this), "PrtFeeSplitExtension must be set");
```

Comment:
Note that in the newer version of [`SnapshotStakingPool.sol`](https://github.com/IndexCoop/periphery/pull/2/files#diff-407efd055ab203d34623f22d840c5a19a919013775c155af86045af703394e4eR36) the public variable is called `distributor` and not `feeSplitExtension`

Reply:
Should we also check to make sure the Staking Pool's `stakeToken` matches the address stored in `prt`?

In the constructor we check if `prt.setToken()` is correct, but I don't think that's effective unless we also check that the StakingPool uses that same prt.

Reply:
added additional checks to match the `SnapshotStakingPool` here [`515e783`](https://github.com/IndexCoop/index-coop-smart-contracts/pull/176/commits/515e7838fad7accf225377ebff80ad194cfdac93)